lib/paper_trail/request.rb

```diff
@@ -0,0 +1,152 @@
+module PaperTrail
+  # Manages the request_store that affects the current request
+  class Request
+    # Sets whether PaperTrail is enabled or disabled for the current request.
+    # @api public
+    def enabled_for_controller=(value)
+      paper_trail_store[:request_enabled_for_controller] = value
+    end
+
+    # Returns `true` if PaperTrail is enabled for the request, `false` otherwise.
+    #
+    # See `PaperTrail::Rails::Controller#paper_trail_enabled_for_controller`.
+    # @api public
+    def enabled_for_controller?
+      !!paper_trail_store[:request_enabled_for_controller]
+    end
+
+    # Sets whether PaperTrail is enabled or disabled for this model in the
+    # current request.
+    # @api public
+    def enabled_for_model(model, value)
+      paper_trail_store[:"enabled_for_#{model}"] = value
+    end
+
+    # Returns `true` if PaperTrail is enabled for this model in the current
+    # request, `false` otherwise.
+    # @api public
+    def enabled_for_model?(model)
+      !!paper_trail_store.fetch(:"enabled_for_#{model}", true)
+    end
+
+    # Sets who is responsible for any changes that occur. You would normally use
+    # this in a migration or on the console, when working with models directly.
+    # In a controller it is set automatically to the `current_user`.
+    # @api public
+    def whodunnit=(value)
+      paper_trail_store[:whodunnit] = value
+    end
+
+    # If nothing passed, returns who is reponsible for any changes that occur.
+    #
+    #   PaperTrail.request.whodunnit = "someone"
+    #   PaperTrail.request.whodunnit # => "someone"
+    #
+    # If value and block passed, set this value as whodunnit for the duration of the block
+    #
+    #   PaperTrail.request.whodunnit("me") do
+    #     puts PaperTrail.request.whodunnit # => "me"
+    #   end
+    #
+    # @api public
+    def whodunnit(value = nil)
+      config_handler(:whodunnit, value) do
+        yield
+      end
+    end
+
+    # Sets any information from the controller that you want PaperTrail to
+    # store.  By default this is set automatically by a before filter.
+    # @api public
+    def controller_info=(value)
+      paper_trail_store[:controller_info] = value
+    end
+
+    # If nothing passed, returns the paper trail info from the controller that you want PaperTrail
+    # to store
+    # See `PaperTrail::Rails::Controller#info_for_paper_trail`.
+    #
+    #   PaperTrail.request.controller_info = { ip: request_user_ip }
+    #   PaperTrail.request.controller_info # => { ip: '127.0.0.1' }
+    #
+    # If value and block passed, set this value as the controller info for the duration of the block
+    #
+    #   PaperTrail.request.controller_info({ ip: '127.0.0.1' }) do
+    #     puts PaperTrail.request.controller_info # => { ip: '127.0.0.1' }
+    #   end
+    #
+    # @api public
+    def controller_info(value = nil)
+      config_handler(:controller_info, value) do
+        yield
+      end
+    end
```

Comment:
I'm not certain we want to add block support to all of these individual config options, given that we have `PaperTrail.request({}) do`

Reply:
Okay -- so you're thinking just leave the `whodunnit` with block support and the have the rest use the old behavior, but can be set with blocks using the new method?  That sounds good to me.  Under the hood I can just have `whodunnit` call the new method as well for readability.